code/game/machinery/vending/vending.dm

```diff
@@ -543,12 +543,23 @@
 /obj/machinery/vending/proc/release_item(datum/data/vending_product/R, delay_vending = 0, dump_product = 0)
 	set waitfor = 0
 	if(delay_vending)
-		use_power(vend_power_usage)	//actuators and stuff
-		if (icon_vend) flick(icon_vend,src) //Show the vending animation if needed
-		sleep(delay_vending)
+		if(powered(power_channel))
```

Comment:
This is an interesting system you are adding for vendors. Tie it with an actual battery and it's golden.

Reply:
Why do I need an actual battery instead of a more efficient variable?

Reply:
It's cleaner and more extendable. In game it also makes a lot more sense if the power supply is shared with other similar devices.